web-common-marlowe/src/Marlowe/Market/Contract5.purs

```diff
@@ -5,197 +5,81 @@ module Marlowe.Market.Contract5
-                                    )
-                                    ( Pay
-                                        (Role "alice")
-                                        (Party (Role "bob"))
-                                        (Token "" "")
-                                        (ConstantParam "amount")
-                                        Close
-                                    )
-                                ]
-                                (SlotParam "arbitrageTimeout")
-                                Close
-                            )
-                        )
-                    ]
-                    (SlotParam "bobTimeout")
-                    ( When
-                        [ Case
-                            ( Choice
-                                ( ChoiceId
-                                    "choice"
-                                    (Role "carol")
-                                )
-                                [ Bound one one ]
-                            )
-                            Close
-                        , Case
-                            ( Choice
-                                ( ChoiceId
-                                    "choice"
-                                    (Role "carol")
-                                )
-                                [ Bound one one ]
-                            )
-                            ( Pay
-                                (Role "alice")
-                                (Party (Role "bob"))
-                                (Token "" "")
-                                (ConstantParam "amount")
-                                Close
-                            )
-                        ]
-                        (SlotParam "arbitrageTimeout")
-                        Close
-                    )
+                ( Deposit
+                    (Role "Dollar provider")
+                    (Role "Dollar provider")
+                    (Token "85bb65" "dollar")
+                    (ConstantParam "Amount of dollars")
                 )
-            ]
-            (SlotParam "aliceTimeout")
-            ( When
-                [ Case
-                    ( Choice
-                        ( ChoiceId
-                            "choice"
-                            (Role "carol")
-                        )
-                        [ Bound one one ]
-                    )
-                    Close
-                , Case
-                    ( Choice
-                        ( ChoiceId
-                            "choice"
-                            (Role "carol")
-                        )
-                        [ Bound zero zero ]
+                ( Pay
+                    (Role "Ada provider")
+                    (Party (Role "Dollar provider"))
+                    (Token "" "")
+                    ( MulValue
+                        (Constant $ BigInteger.fromInt 1000000)
```

Comment:
Eventually this should be an oracle, right? We probably need a way to display this to the users somehow 🤔

Reply:
Maybe, yes, I am not sure how oracles are going to work on the prototype, nor tokens, we need to discuss this later and we can update the examples then